src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/SqlAuthenticationProvider.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Data.SqlClient;
+
+/// <include file='../doc/SqlAuthenticationProvider.xml' path='docs/members[@name="SqlAuthenticationProvider"]/SqlAuthenticationProvider/*'/>
+public abstract class SqlAuthenticationProvider
+{
+    /// <include file='../doc/SqlAuthenticationProvider.xml' path='docs/members[@name="SqlAuthenticationProvider"]/BeforeLoad/*'/>
+    public virtual void BeforeLoad(SqlAuthenticationMethod authenticationMethod) { }
+
+    /// <include file='../doc/SqlAuthenticationProvider.xml' path='docs/members[@name="SqlAuthenticationProvider"]/BeforeUnload/*'/>
+    public virtual void BeforeUnload(SqlAuthenticationMethod authenticationMethod) { }
+
+    /// <include file='../doc/SqlAuthenticationProvider.xml' path='docs/members[@name="SqlAuthenticationProvider"]/IsSupported/*'/>
+    public abstract bool IsSupported(SqlAuthenticationMethod authenticationMethod);
+
+    /// <include file='../doc/SqlAuthenticationProvider.xml' path='docs/members[@name="SqlAuthenticationProvider"]/AcquireTokenAsync/*'/>
+    public abstract Task<SqlAuthenticationToken> AcquireTokenAsync(SqlAuthenticationParameters parameters);
+}
```

Comment:
Add new sync API? (Or maybe you planned that for a little later...)
```suggestion
}

    public virtual SqlAuthenticationToken AcquireToken(SqlAuthenticationParameters parameters) => Task.Run(async () => await AcquireTokenAsync(parameters)).GetAwaiter().GetResult();
```

Reply:
Yes, in a future PR.